src/main/java/org/shanerx/tradeshop/listeners/ShopTradeListener.java

```diff
@@ -69,8 +68,13 @@ public void onBlockInteract(PlayerInteractEvent e) {
         JsonConfiguration json = new JsonConfiguration(s.getChunk());
         shop = json.loadShop(new ShopLocation(s.getLocation()));
 
-        if (shop == null)
+        if (shop == null) {
+            s.setLine(0, "");
+            s.setLine(1, "");
+            s.setLine(2, "");
+            s.setLine(3, "");
```

Comment:
If a shop doesnt exist I am clearing the sign since it isnt linked to a shop, I did forget to update the sign tho, will add that in

Reply:
mhm.. and under which circumstances would this

```
if (ShopType.isShop(e.getClickedBlock())) {
            s = (Sign) e.getClickedBlock().getState();
        }
```

run, and yet the sign not be connected to a valid shop?

Reply:
XANT3R had an error come up where the shop sign existed but there was no shop in the config, which this prevents, should also clear the sign to prevent it from continuing to happen